packages/vue-components/src/Navbar.vue

```diff
@@ -71,12 +75,103 @@
         default:
           return 'navbar-dark bg-primary'
         }
-      }
+      },
     },
     methods: {
       toggleCollapse (e) {
         e && e.preventDefault()
         this.collapsed = !this.collapsed
+      },
+      normalizeUrl(url) {
+        if (url.endsWith('.html')) {
+          return url.toLowerCase();
+        } else if (url.endsWith('/')) {
+          return `${url}index.html`.toLowerCase();
+        } else if (!url.endsWith('/')) {
+          return `${url}.html`.toLowerCase();
+        } else {
+          return url.toLowerCase();
+        }
+      },
+      // Splits a normalised URL into its parts, e.g http://site.org/foo/bar/index.html -> ['foo','bar','index.html']
+      splitUrl(url) {
+        const u = new URL(this.normalizeUrl(url));
+        return `${u.pathname}`.substr(1).split('/');
+      },
+      isSibling(url, href) {
+        const hParts = this.splitUrl(href);
+        const uParts = this.splitUrl(url);
+        if (hParts.length !== uParts.length) {
+          return false;
+        }
+        for (let i = 0; i < hParts.length - 1; i++) {
+          if (hParts[i] !== uParts[i]) {
+            return false;
+          }
+        }
+        return true;
+      },
+      isChild(url, href) {
+        const hParts = this.splitUrl(href);
+        const uParts = this.splitUrl(url);
+        if (hParts.length <= uParts.length) {
+          return false;
+        }
+        for (let i = 0; i < uParts.length - 1; i++) {
+          if (hParts[i] !== uParts[i]) {
+            return false;
+          }
+        }
+        return true;
+      },
+      highlightLink(url) {
+        const hlMode = this.highlightOn;
+        if (hlMode === 'none') {
+          return;
+        }
+        const navLis = Array.from(this.$el.querySelector('.navbar-nav').children);
+        // attempt an exact match first
+        for (const li of navLis) {
+          const navLinks = Array.from(li.querySelectorAll('a.nav-link'));
+          const dropdownLinks = Array.from(li.querySelectorAll('a.dropdown-item'));
+          const allNavLinks = navLinks.concat(dropdownLinks).filter(a => a.href);
+          for (const a of allNavLinks) {
+            const aNorm = this.normalizeUrl(a.href);
+            const urlNorm = this.normalizeUrl(url);
+            // terminate early on an exact match
+            if (aNorm === urlNorm) {
+              li.classList.add('current');
+              return;
+            }
+          }
+        }
+        // fallback to user preference, otherwise
+        for (const li of navLis) {
+          const navLinks = Array.from(li.querySelectorAll('a.nav-link'));
+          const dropdownLinks = Array.from(li.querySelectorAll('a.dropdown-item'));
+          const allNavLinks = navLinks.concat(dropdownLinks).filter(a => a.href);
+          for (const a of allNavLinks) {
+            if (hlMode === 'sibling-or-child') {
```

Comment:
in reference to https://github.com/MarkBind/markbind/pull/1379#discussion_r521406793, what needs to be checked is `<a highlight-on> attribute || this.highlightMode`

would need an additional if clause for `none` here as well